src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -230,27 +267,62 @@ export class StudyViewPageStore {
                     break;
                 default:
                     this._clinicalDataEqualityFilterSet.delete(chartMeta.uniqueKey);
+                    this._clinicalDataIntervalFilterSet.delete(chartMeta.uniqueKey);
                     break;
             }
         }
         this.changeChartVisibility(chartMeta.uniqueKey, visible);
     }
 
+    @action
+    toggleLogScale(chartMeta: ChartMeta) {
+        // reset filters before toggling
+        this.updateClinicalDataIntervalFilters(chartMeta, []);
+
+        // then toggle
+        if (this.logScaleState.get(chartMeta.uniqueKey) === undefined) {
+            this.logScaleState.set(chartMeta.uniqueKey, false);
+        }
+        else {
+            this.logScaleState.set(chartMeta.uniqueKey, !this.logScaleState.get(chartMeta.uniqueKey));
+        }
+    }
+
+    public isLogScaleToggleVisible(uniqueKey: string, dataBins?: DataBin[]) {
+        return (
+            this.logScaleState.get(uniqueKey) !== undefined ||
+            isLogScaleByDataBins(dataBins)
+        );
+    }
+
+    public isLogScaleDisabled(uniqueKey: string) {
+        return this.logScaleState.get(uniqueKey) === false;
+    }
+
+    public isLogScaleChecked(uniqueKey: string) {
+        return this.logScaleState.get(uniqueKey) !== false;
+    }
+
     @computed private get emptyFilter(): StudyViewFilter {
         return { studyIds: this.studyIds } as any;
     }
 
     @computed
     get filters() {
-        let filters: StudyViewFilter = {} as any;
+        const filters: StudyViewFilter = {} as any;
```

Comment:
Maybe Partial<StudyViewFilter> instead of any?

Reply:
changed `any` to `Partial`